pkg/controller/builder.go

```diff
@@ -102,10 +104,27 @@ func (b *Builder) BuildControllers(subDirectories ...string) (map[string]factory
 				return nil
 			}
 
+			if manifestGVK.Kind == "ClusterCatalog" && manifestGVK.Group == "olm.operatorframework.io" {
+				controllerName := controllerNameForObject(namePrefix, &manifest)
```

Comment:
Any reason we need an individual controller for each `ClusterCatalog`? Seems like we should be able to have a single controller that handles all of them.

Reply:
We certainly could have a single controller that handles all of them. I'm not _super_ opinionated here, but my thinking for individual controller was that we have:
- Slightly easier to understand logic in the controller because we don't have to loop through all of them and have special logic for handling failure to keep resources updated in the middle of the loop
- More nuanced signal for which `ClusterCatalog` resources are degraded because each controller gets it's own `Degraded` condition. If we went with a single controller any failure would result in a `Degraded` signal that requires parsing the string message to understand which `ClusterCatalog` resources are actually degraded. Those that care about the health of individual `ClusterCatalog` resources could build better automation against the individual degraded conditions.
- Maintains consistency with the `DeploymentController` that we use for specific resources (although this is the least important point IMO)

I felt that these pros outweighed the con of having more controllers running, but provided a strong argument for having a single controller that proves otherwise I'm happy to change this.

Reply:
Left as is based on a synchronous discussion.